src/cdk/stepper/stepper.ts

```diff
@@ -84,7 +97,8 @@ export class CdkStepper {
   @Input()
   get selectedIndex() { return this._selectedIndex; }
   set selectedIndex(index: number) {
-    if (this._selectedIndex != index) {
+    this._steps.toArray()[this._selectedIndex].interacted = true;
+    if (this._selectedIndex != index && !this._steps.toArray()[index].disabled) {
       this._emitStepperSelectionEvent(index);
```

Comment:
I thought this is commonly called by all user interactions and that's why we decided to delegate emitting a change event here. Is this not what you meant?